### PR TITLE
HDDS-11124. Removed DELETED_TABLE and DELETED_DIR_TABLE locks

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
As part of the investigation of HDDS-11124, it was found that change HDDS-8067 created a deadlock and caused the read/write operations to halt.

<ins>**Deadlock scenario**</ins>: Handler threads waiting for applyTransaction future -> which is waiting for DoubleBuffer which is in progress -> DoubleBuffer flush thread is stuck on processing a SnapshotCreate request -> because OmSnapshotManager is waiting for the DeletedTable lock -> which is being held by KeyDeletingService -> which is waiting for deletion to be applied by OzoneManagerStateMachine#applyTransaction.
More detailed analysis in HDDS-11124.

As part of this change, DELETED_TABLE and DELETED_DIR_TABLE locks are removed and we are relying on transaction flush which happens sequentially on both Ratis and DoubleBuffer for now.

## What is the link to the Apache JIRA
HDDS-11124

## How was this patch tested?
Existing tests.

### CODE REVIEW

File: hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/OMMetadataManager.java [hunk 1]
```diff
@@ -322,13 +321,6 @@ ExpiredOpenKeys getExpiredOpenKeys(Duration expireThreshold, int count,
   List<ExpiredMultipartUploadsBucket> getExpiredMultipartUploads(
       Duration expireThreshold, int maxParts) throws IOException;
 
-  /**
-   * Retrieve RWLock for the table.
-   * @param tableName table name.
-   * @return ReentrantReadWriteLock
-   */
-  ReentrantReadWriteLock getTableLock(String tableName);
```
Comment (smengcl) [inline hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/OMMetadataManager.java:330]: Need to adjust Recon Impl as well:

```
Error:  COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
Error:  /home/runner/work/ozone/ozone/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/recovery/ReconOmMetadataManagerImpl.java:[75,3] method does not override or implement a method from a supertype
Error:  /home/runner/work/ozone/ozone/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/recovery/ReconOmMetadataManagerImpl.java:[77,17] cannot find symbol
  symbol: method getTableLock(java.lang.String)
```

**Review by smengcl — APPROVED** [review]
lgtm. I have one minor comment inline.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java [hunk 2]
```diff
@@ -453,44 +453,26 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
 
     final DBCheckpoint dbCheckpoint;
 
-    // Acquire active DB deletedDirectoryTable write lock to block
-    // DirDeletingTask
-    omMetadataManager.getTableLock(OmMetadataManagerImpl.DELETED_DIR_TABLE)
-        .writeLock().lock();
-    // Acquire active DB deletedTable write lock to block KeyDeletingTask
-    omMetadataManager.getTableLock(OmMetadataManagerImpl.DELETED_TABLE)
-        .writeLock().lock();
```
[the remark this answers is not in the document]
Comment (hemantk-12) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java:462]: I updated the comment here: https://github.com/apache/ozone/pull/6921/files#diff-857b24769ac9c1c67d015bdffc1227dc2b4751de3790b9fd89269a3c46a5218aR469

Does it make sense? Or I should add a comment here explicitly?

[the remark this answers is not in the document]
Comment (smengcl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java:462]: Yeah it works. Pls feel free to merge this.

**Pull request merged**